Content.Shared/Pinpointer/SharedPinpointerSystem.cs

```diff
@@ -27,13 +34,42 @@ private void OnAfterInteract(EntityUid uid, PinpointerComponent component, After
         if (!args.CanReach || args.Target is not { } target)
             return;
 
-        if (!component.CanRetarget || component.IsActive)
-            return;
-
         // TODO add doafter once the freeze is lifted
         args.Handled = true;
-        component.Target = args.Target;
-        _adminLogger.Add(LogType.Action, LogImpact.Low, $"{ToPrettyString(args.User):player} set target of {ToPrettyString(uid):pinpointer} to {ToPrettyString(component.Target.Value):target}");
+
+        if (component.CanRetarget && !component.IsActive || _tagSystem.HasTag(args.Target.Value, "PinpointerScannable"))
+        {
+            component.Target = args.Target;
+        }
+
+        else if (TryComp<NukeCodePaperComponent>(args.Target, out var nukeCodePaperComponent))
+        {
+            if (nukeCodePaperComponent.Nuke == null)
+                return;
+
+            component.Target = nukeCodePaperComponent.Nuke;
+        }
+        else
+        {
+            return;
+        }
+
+        if(component.StoredTargets.Contains(component.Target.Value))
+            return;
+
+        if (component.StoredTargets.Count >= component.MaxTargets && _net.IsServer)
+        {
+            _popup.PopupEntity(Loc.GetString("target-pinpointer-full"),args.User,args.User);
+            return;
+        }
+
+        component.StoredTargets.Add(component.Target.Value);
+        if (_net.IsServer)
+        {
+            _popup.PopupEntity(Loc.GetString("target-pinpointer-stored", ("target", component.Target.Value)), args.User, args.User);
+            _adminLogger.Add(LogType.Action, LogImpact.Low, $"{ToPrettyString(args.User):player} set target of {ToPrettyString(uid):pinpointer} to {ToPrettyString(component.Target.Value):target}");
+        }
+
```

Comment:
Client is going to keep adding targets indefinitely so I don't understand why this is being done like this? Additionally this is going to spam admin logs; pinpointer targets are irrelevant.

Reply:
I also don't know why I did it like that.
Admin logger was already in the code so I just left it in there.